src/env.cc

```diff
@@ -262,6 +262,12 @@ bool Environment::EmitNapiWarning() {
   return current_value;
 }
 
+bool Environment::EmitDomainWarning() {
```

Comment:
Not a great name.  The imperative suggests it performs an action but it's really a getter, although one with side effects (which itself is somewhat objectionable.)

At the very least call it something like `ShouldEmitDomainWarning()`, and look into separating the check and the update.  Since you're only using it in one place, I'd split it in a getter and setter and call the setter explicitly after emitting the warning.

Reply:
Hehe. Basically just copied `EmitNapiWarning` :D